test/unittest/Database/DatabaseEraseTests.cpp

```diff
@@ -0,0 +1,201 @@
+        }
+        for (auto writer : writers)
+        {
+            EXPECT_EQ(locator.second->data_writers.find(writer->id), locator.second->data_writers.end());
+        }
+    }
+    // Any reference to the erased domain has been deleted
+    for (auto process : db.domains_by_process())
+    {
+        EXPECT_EQ(process.second.find(domain_id), process.second.end());
+    }
+    EXPECT_EQ(db.processes_by_domain().find(domain_id), db.processes_by_domain().end());
+    // Any reference to the erased participant has been deleted
+    for (auto participant : participants)
+    {
+        EntityId participant_id = participant->id;
+        for (auto process : db.processes())
+        {
+            EXPECT_EQ(process.second->participants.find(participant_id), process.second->participants.end());
+        }
+        for (auto locator : db.participants_by_locator())
+        {
+            EXPECT_EQ(locator.second.find(participant_id), locator.second.end());
+        }
+        EXPECT_EQ(db.locators_by_participant().find(participant_id), db.locators_by_participant().end());
+    }
+}
+
+/**
+ * This test checks the erase feature.
+ * It loads the complex_dump.json and erases the domain_1.
+ * The resulting dump should be equal to the one loaded from complex_dump_erased_domain_1.json.
+ */
+TEST(database_erase_tests, erase_domain)
+{
+    // Read JSON files
+    DatabaseDump initial_dump;
+    DatabaseDump final_dump;
+    load_file(COMPLEX_DUMP_FILE, initial_dump);
+    load_file(COMPLEX_ERASED_DUMP_FILE, final_dump);
+
+    // Create database
+    DataBaseTest db;
+
+    // Load initial dump in the database
+    db.load_database(initial_dump);
+
+    // Call erase monitor removing domain_1
+    std::vector<std::pair<EntityId, EntityId>> domains = db.get_entities_by_name(EntityKind::DOMAIN, "domain_1");
+    EntityId domain_id = domains.begin()->first;
+    // Save entities associated to the erased domain to check that the cross maps are correctly erased.
+    std::vector<std::shared_ptr<const Entity>> participants = db.get_entities(EntityKind::PARTICIPANT, domain_id);
+    std::vector<std::shared_ptr<const Entity>> readers = db.get_entities(EntityKind::DATAREADER, domain_id);
+    std::vector<std::shared_ptr<const Entity>> writers = db.get_entities(EntityKind::DATAWRITER, domain_id);
+    db.erase(domain_id);
+
+    check_erased_database(db, domain_id, participants, readers, writers);
+
+    // Calling again to erase an already erased domain throws an exception
+    ASSERT_THROW(db.erase(domains.begin()->first), BadParameter);
+
+    // Dump erased database
+    DatabaseDump erased_dump = db.dump_database();
+
+    // Compare erased and final dumps
+    EXPECT_EQ(final_dump, erased_dump);
+}
+
+// This test checks that erasing a database where the participant is not yet linked to the process works as expected
+TEST(database_erase_tests, erase_domain_unlinked_participant_process)
```

Comment:
This is the same as `erase_domain` but with different dumps. We can extract this to a function that takes the dump names and make the tests call this function instead